packages/amplify-graphql-schema-test-library/src/index.ts

```diff
@@ -0,0 +1,131 @@
+  transformerVersion: TransformerVersion;
+  supportedPlatforms: TransformerPlatform;
+  sdl: string;
+};
+
+export const schemas: { [key: string]: TransformerSchema } = {
+  '@model-simple': {
+    description: 'Simple @model schema',
+    transformerVersion: TransformerVersion.all,
+    supportedPlatforms: TransformerPlatform.all,
+    sdl: `
+      type Todo @model {
+        id: ID!
+        name: String!
+      }
+    `,
+  },
+  'v2-primary-key-with-composite-sort-key': {
+    description: '@primaryKey with a composite sort key',
+    transformerVersion: TransformerVersion.v2,
+    supportedPlatforms: TransformerPlatform.all,
+    sdl: `
+      type Test @model {
+        email: String! @primaryKey(sortKeyFields: ["kind", "other"])
+        kind: Int!
+        other: AWSDateTime!
+        yetAnother: String
+        andAnother: String!
+      }
+    `,
+  },
+  'v2-index-with-queryfield': {
+    description: '@index with queryField',
+    transformerVersion: TransformerVersion.v2,
+    supportedPlatforms: TransformerPlatform.all,
+    sdl: `
+      type Test @model {
+        email: String!
+        createdAt: AWSDateTime!
+        category: String! @index(name: "CategoryGSI", sortKeyFields: "createdAt", queryField: "testsByCategory")
+        description: String
+      }
+    `,
+  },
+  'v2-recursive-has-one-dependency': {
+    description: 'Recursive @hasOne relationship',
+    transformerVersion: TransformerVersion.v2,
+    supportedPlatforms: TransformerPlatform.all,
+    sdl: `
+      type Item @model {
+        id: ID!
+        item: Item @hasOne
+      }
+    `,
+  },
+  'v2-recursive-has-many-dependency': {
+    description: 'Recursive @hasMany relationship',
+    transformerVersion: TransformerVersion.v2,
+    supportedPlatforms: TransformerPlatform.all,
+    sdl: `
+      type Item @model {
+        id: ID!
+        items: [Item] @hasMany
+      }
+    `,
+  },
+  'v2-cyclic-has-one-dependency': {
+    description: 'Cyclic @hasOne dependency between two models',
+    transformerVersion: TransformerVersion.v2,
+    supportedPlatforms:
+      TransformerPlatform.api |
+      TransformerPlatform.js |
+      TransformerPlatform.ios |
+      TransformerPlatform.android |
+      TransformerPlatform.flutter,
+    sdl: `
+      type Blog @model {
+        id: ID!
+        posts: Post @hasOne
+      }
+      type Post @model {
+        id: ID!
+        blog: Blog @hasOne
+      }
+    `,
+  },
+  'v2-cyclic-has-many-dependency': {
+    description: 'Cyclic @hasMany dependency between two models',
+    transformerVersion: TransformerVersion.v2,
+    supportedPlatforms:
+      TransformerPlatform.api |
+      TransformerPlatform.js |
+      TransformerPlatform.ios |
+      TransformerPlatform.android |
+      TransformerPlatform.flutter,
+    sdl: `
+      type Blog @model {
+        id: ID!
+        posts: [Post] @hasMany
+      }
+      type Post @model {
+        id: ID!
+        blog: [Blog] @hasMany
+      }
+    `,
```

Comment:
I think adding an default value use-case would be valuable.

```
type Todo @model {
  content: String @default(value: "My new Todo")
}
```

implicit and explicit multi-model has one and has many, so:

```
type Project @model {
  id: ID!
  name: String
  team: Team @hasOne
}

type Team @model {
  id: ID!
  name: String!
}
```

```
type Project @model {
  id: ID!
  name: String
  teamID: ID
  team: Team @hasOne(fields: ["teamID"])
}

type Team @model {
  id: ID!
  name: String!
}
```

```
type Post @model {
  id: ID!
  title: String!
  comments: [Comment] @hasMany
}

type Comment @model {
  id: ID!
  content: String!
}
```

```
type Post @model {
  id: ID!
  title: String!
  comments: [Comment] @hasMany(indexName: "byPost", fields: ["id"])
}

type Comment @model {
  id: ID!
  postID: ID! @index(name: "byPost", sortKeyFields: ["content"])
  content: String!
}
```

As well as adding in the usecases for belongsTo

```
type Project @model {
  id: ID!
  name: String
  team: Team @hasOne
}

type Team @model {
  id: ID!
  name: String!
  project: Project @belongsTo
}
```

```
type Project @model {
  id: ID!
  name: String
  team: Team @hasOne
}

type Team @model {
  id: ID!
  name: String!
  projectID: ID
  project: Project @belongsTo(fields: ["projectID"])
}
```

```
type Post @model {
  id: ID!
  title: String!
  comments: [Comment] @hasMany(indexName: "byPost", fields: ["id"])
}

type Comment @model {
  id: ID!
  postID: ID! @index(name: "byPost", sortKeyFields: ["content"])
  content: String!
  post: Post @belongsTo(fields: ["postID"])
}
```